File: common/lib/dependabot/requirements_update_strategy.rb
# typed: strong
# frozen_string_literal: true

module Dependabot
  class RequirementsUpdateStrategy < T::Enum
    enums do
      BumpVersions = new("bump_versions")
      BumpVersionsIfNecessary = new("bump_versions_if_necessary")
      LockfileOnly = new("lockfile_only")
      WidenRanges = new("widen_ranges")
    end

    extend T::Sig

    sig { returns(T::Boolean) }
    def lockfile_only?
      self == LockfileOnly
    end
  end
end

File: nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/ApiModel/RequirementsUpdateStrategy.cs
using System.Text.Json.Serialization;

namespace NuGetUpdater.Core.Run.ApiModel;

public enum RequirementsUpdateStrategy
{
    [JsonStringEnumMemberName("bump_versions")]
    BumpVersions,
    [JsonStringEnumMemberName("bump_versions_if_necessary")]
    BumpVersionsIfNecessary,
    [JsonStringEnumMemberName("lockfile_only")]
    LockfileOnly,
    [JsonStringEnumMemberName("widen_ranges")]
    WidenRanges,
}
